### DISCUSSION TITLE
Sensory expiry alert- Medtronic

### DISCUSSION THREAD

**Discussion by chelseabuttafli** [NightscoutFoundation/xDrip#2690] [Q&A]
I turned on the new sensor expiry alert feature before I started my most recent sensor. It has just expired with no alerts at all. I have the sensor expiry setting turned on, it is showing the correct start date and time for the sensor, it is showing sensor expired on the graph and the hardware data source is set to carelink follower. Is there anything I have done wrong? How can I get it to work please? 
![inbound961296271069059410](https://user-images.githubusercontent.com/62456364/220252746-12690852-4e94-4060-9d9e-20444b0c6814.jpg)

**Comment by jamorham** [comment]
I don't think we actually know when a sensor will expire with this data collector. The sensor start date is just when the first reading is received and the items shown on the graph are just processed events. It might be possible but it isn't supported at the moment.

**Reply by chelseabuttafli** [reply]
That's a bit disappointing that it doesn't seem to support one of the major CGMs....

**Comment by jamorham** [comment]
@benceszasz Do you think data for sensor expiry is available?

**Reply by chelseabuttafli** [reply]
I don't know how to tell. That's why I was directed to ask here

**Reply by Navid200** [reply]
@chelseabuttafli That question was for someone else.  Even though you were not tagged for that question, but, you will receive a notification for all posts in this thread.  You can change that and stop receiving notifications.  But, I don't think you would want that because I think you would want to see all the conversation.  But, it's up to you.

**Reply by chelseabuttafli** [reply]
Thank you Navid200. I'm sorry I've never used this platform before

**Reply by benceszasz** [reply]
@jamorham sensor remain time in minutes is provided by CareLink, which is displayed in my private version (extra status line -> hour icon). So yes, using sensor remain time the sensor expiry can be calculated.

**Reply by chelseabuttafli** [reply]
@benceszasz so the version I'm currently using isn't compatible and I need to use your carelink version for the sensor expiry function to work properly?

**Reply by benceszasz** [reply]
@chelseabuttafli no, the sensor expiry alarm is a new function in the central xdrip version, it is not in my old private version. In my private version only the sensor remaining time is displayed, but it is not used in any alarms.

**Reply by chelseabuttafli** [reply]
@benceszasz do you know if any version which WILL alarm for sensor expiry for a medtronic?

**Reply by benceszasz** [reply]
@chelseabuttafli no

**Reply by jamorham** [reply]
@benceszasz how easy is it to get access to this expiry information? Could the collector store it in a persistent store or similar that I could use it in the SensorDays class?

**Reply by chelseabuttafli** [reply]
@jamorham in all honesty I don't really know what you mean. The carelink app tells me exactly how long the sensor has left, the follower app tells me how many days. Benceszasz's carelink version of xdrip does accurately display time left for the sensor but apparently can't do expiry alerts. I can provide screenshots of any of those if that will help?

**Reply by jamorham** [reply]
@chelseabuttafli there is no need to respond to items in the discussion directed at other users via their username and the @ symbol. Those questions are directed at others. In this case @benceszasz who developed the carelink feature and knows how it works. I developed the sensor expiry notification feature and we are discussing if/how these two things can be connected together which is the basis for the issue you are reporting.

**Reply by chelseabuttafli** [reply]
Sorry.. Still learning this platform...

**Reply by benceszasz** [reply]
@jamorham yes, it can be done, the question is where do you need it and how should we handle special cases: sensor is not used for some reason (maybe the user run out of sensors, but still uses the pump), sesnor is in warmup phase, etc.

**Comment by chelseabuttafli** [comment]
Carelink follower xdrip version 
![Screenshot_20230222-222816_xDrip+](https://user-images.githubusercontent.com/62456364/220607858-e488e8aa-e08c-41a6-8cba-cd0c0a2c496f.jpg)

**Reply by chelseabuttafli** [reply]
@benceszasz what does the red triangle reading in the middle mean? I haven't been able to figure that one out...

**Reply by benceszasz** [reply]
@chelseabuttafli this should be Umax (maximum autobasal / hour). For 780G this seems to be correct, but in case of 770G I have received responses, that it doesn't seem to be correct. If you have any questions, issues with my CareLink xdrip version please raise it in my repository, this repository belongs to the main xdrip app.
